cfpb/cfgov-lighthouse
# Add Lighthouse report history pages

Adds permalinks for each page w/ links to their historical results. It does so by reading all the reports in the `docs/reports/` directory and generating a JSON index of all the report data organized by both dates and pages. Webpack reads the JSON file and spits out an HTML file for each page.

Note: Ignore all the changes in the `docs/` directory. In a future PR we'll move all those files to their own branch and gitignore the directory.

## Changes

- Modified `index.njk` to accommodate the new data format and added `page.njk` for the new pages.
- Added scripts under `/scripts` for processing the reports. The entry point is `scripts/create-reports-index.js`.
- Tests in `test/` to validate the above scripts.

## Testing

1. Pull down this branch
1. `yarn` to install dependencies.
1. `yarn test` to ensure tests pass.
1. `yarn index-reports` to index all the reports in docs/reports/ and generate docs/reports.json. It will also delete extraneous report files.
1. `yarn build` to run webpack and generate the website (homepage and individual cf.gov pages) from reports.json.
1. `yarn serve` to server the docs/ directory locally and preview the site at http://127.0.0.1:8080/.
1. Click the "View report history" links to view individual pages' histories.

## Screenshots

<img width="1080" alt="Screen Shot 2020-10-14 at 2 40 28 PM" src="https://user-images.githubusercontent.com/1060248/96031389-45a8b900-0e2b-11eb-9b13-3427a585d355.png">

## Notes

-

## Todos

-

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
